### PR TITLE
Cult structures take damage from melee and projectiles.

### LINKED ISSUE #7699: Cult structures cannot be deconstructed
**Problem Description**:
Cult structures cannot be deconstructed.

**What did you expect to happen**: 
I expected there to be a way to remove them, even if it could only be done by the chaplain.

**What happened instead**:
Briefly looking over the code, I didn't see any deconstruction code that the crew could use.

**Why is this bad/What are the consequences:**
While destroying runes is simple enough, destroying the cult structures should be possible as well.

**Steps to reproduce the problem**:
Create a daemon forge. Try all sorts of ways to remove the structure.

### PR DESCRIPTION
**Pull request by Birdtalon**
This PR makes cult structures susceptible to damage from melee and from projectiles.

Now you can destroy them with something other than bombs.

Fixes #7699 

🆑 Birdtalon
add: Cultist structures can now be destroyed using melee and or projectiles
/🆑

### CODE REVIEW

**Comment by Aurorablade** [comment]
It's not good to just snow flake it on here. If we are gonna do this we need to finish it from here https://github.com/ParadiseSS13/Paradise/pull/6962 but it's also going to invoice a refactor of our attack hand code

**Comment by Birdtalon** [comment]
It's been months since the new cult was added onto the server and the inability to destroy any of the cult structures without bombs is debilitating to security when dealing with cult bases. If someone is actually going to complete the porting of the /tg/ obj damage system then sure. But we're now in August with no sign of it. If it actually happens then reverting these 20 or so lines will be child's play.

Or we could just ignore it being a problem and hope someone finishes that port and refactor.

**Comment by Tayyyyyyy** [comment]
This doesn't look snowflakey at all. I think it's a very good stopgap for now. The inability to destroy cult structures is really problematic. I remember one round where we just got trapped into a bunch of cult structures. It gives you no chance for counterplay. You're just out of the round at that point.

**Comment by scrubmcnoob** [comment]
Yes, please.

**Comment by Fox-McCloud** [comment]
@Aurorablade it's not ideal, no, but the amount of work required to implement this *properly* is astounding and unlikely to be completed in the next week or two, so...this is preferable *for now*.

File: code/game/gamemodes/cult/cult_structures.dm [hunk 1]
```diff
@@ -65,6 +65,23 @@
...
+
+/obj/structure/cult/functional/proc/take_damage(damage, damage_type = BRUTE)
+	if(damage_type != STAMINA)
+		health -= damage
+		updatehealth()
+
+/obj/structure/cult/functional/attackby(obj/item/I, mob/living/user)
+	..()
+	take_damage(I.force * 0.5, I.damtype)
+	playsound(loc, 'sound/effects/grillehit.ogg', 80, 1)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/cult_structures.dm]: should probably play the sound that the object that it it makes.

File: code/game/gamemodes/cult/cult_structures.dm [hunk 2]
```diff
@@ -65,6 +65,23 @@
 	else
 		return ..()
 
+/obj/structure/cult/functional/proc/updatehealth()
+	if(health <= 0)
+		destroy_structure()
+
+/obj/structure/cult/functional/proc/take_damage(damage, damage_type = BRUTE)
+	if(damage_type != STAMINA)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/cult_structures.dm]: only `BRUTE` and `BURN` should work.

I don't think an oxygen damaging or clone damaging bullet should destroy this.

File: code/game/gamemodes/cult/cult_structures.dm [hunk 3]
```diff
@@ -65,6 +65,23 @@
...
+		health -= damage
+		updatehealth()
+
+/obj/structure/cult/functional/attackby(obj/item/I, mob/living/user)
+	..()
+	take_damage(I.force * 0.5, I.damtype)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/cult_structures.dm]: Don't halve this.

File: code/game/gamemodes/cult/cult_structures.dm (hunk 1 above)
Comment (Birdtalon) [inline code/game/gamemodes/cult/cult_structures.dm]: Done.

File: code/game/gamemodes/cult/cult_structures.dm (hunk 2 above)
Comment (Birdtalon) [inline code/game/gamemodes/cult/cult_structures.dm]: Is there a cleaner way than doing `if(damage_type == BURN || damage_type == BRUTE`?

File: code/game/gamemodes/cult/cult_structures.dm (hunk 3 above)
Comment (Birdtalon) [inline code/game/gamemodes/cult/cult_structures.dm]: Updated.

**Comment by Birdtalon** [comment]
Updated with your requests @Fox-McCloud

**Comment by tigercat2000** [comment]
This will do for now.

**Pull request merged**